apps/api/e2e/utils/server.ts

```diff
@@ -18,167 +18,64 @@ const BASE_URL = `http://${HOST}:${PORT}${API_PREFIX ? `/${API_PREFIX}` : ""}`;
  *
  * @returns A promise that resolves to the HTTP server instance
  */
-export async function startServer(): Promise<Server> {
-  // First try to kill any existing servers on the test port
-  await killExistingServers();
+export async function startServer(): Promise<void> {
+  if (serverProcess) {
+    throw new Error("Cannot start test server twice");
+  }
 
-  return new Promise((resolve, reject) => {
-    try {
-      const testPort = process.env.TEST_PORT || "3001";
-      const testHost = process.env.TEST_HOST || "0.0.0.0"; // Bind to all interfaces in Docker
-
-      console.log(`Starting test server on ${testHost}:${testPort}...`);
-
-      // Start the server script in a separate process
-      // Use the index.ts file which is the main entry point
-      serverProcess = spawn("npx", ["tsx", "src/index.ts"], {
-        env: {
-          ...process.env,
-          NODE_ENV: "test",
-          PORT: testPort,
-          HOST: testHost,
-          METRICS_PORT: process.env.METRICS_PORT || "3003",
-          METRICS_HOST: "127.0.0.1", // Secure binding for tests
-          TEST_MODE: "true",
-        },
-        stdio: "inherit",
-        detached: true,
```

Comment:
By removing the detached option, we can now just do `ctrl+c` and actually stop the e2e tests.